envoy/common/inline_map_registry.h

```diff
@@ -0,0 +1,513 @@
+#pragma once
+
+#include <cstddef>
+#include <cstdint>
+
+#include "source/common/common/assert.h"
+#include "source/common/common/macros.h"
+#include "source/common/common/utility.h"
+
+#include "absl/container/flat_hash_map.h"
+#include "absl/container/node_hash_set.h"
+
+namespace Envoy {
+
+class UntypedInlineHandle {
+public:
+  UntypedInlineHandle(uint64_t inline_scope_id, uint64_t inline_entry_id,
+                      absl::string_view inline_entry_key)
+      : inline_scope_id_(inline_scope_id), inline_entry_id_(inline_entry_id),
+        inline_entry_key_(inline_entry_key) {}
+
+  uint64_t inlineScopeId() const { return inline_scope_id_; }
+
+  uint64_t inlineEntryId() const { return inline_entry_id_; }
+
+  absl::string_view inlineEntryKey() const { return inline_entry_key_; }
+
+private:
+  uint64_t inline_scope_id_{};
```

Comment:
It a unique id to represent a scope of inline map registry and it is allocated by the InlineMapRegistryManager. It's necessary for the untyped inline handle. But it's ok for me to remove it for now because it's optional feature.